sdk/identity/azure_identity/src/process_ext.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// cspell:ignore workdir
+
+use azure_core::{
+    credentials::AccessToken,
+    error::{Error, ErrorKind, Result},
+    process::Executor,
+};
+use std::{ffi::OsStr, marker::PhantomData};
+
+use crate::env::Env;
+
+#[async_trait::async_trait]
+pub trait ExecutorExt {
```

Comment:
1. Why is it public?
2. Why not just a simple function? Since I see no reason this should be publicly exported from a crate that is for authentication and because this doesn't seem to be used in a way that benefits from extension methods, just make it a normal function. Simplifies design.

Reply:
1. It isn't? `lib.rs` doesn't `pub use` it. It definitely shouldn't be exposed to applications though; this should be `pub(crate)`
2. I anticipate making this a normal method on `Executor` when we move that type to this crate